enterprise/models.py

```diff
@@ -1579,10 +1578,9 @@ def get_course(self, course_key):
         Return:
             dict: The course metadata.
         """
-        request = crum.get_current_request()
-        # Temporarily gate enterprise catalog api usage behind waffle flag
-        if request and waffle.flag_is_active(request, USE_ENTERPRISE_CATALOG):
-            if not EnterpriseCatalogApiClient(user=request.user).contains_content_items(self.uuid, [course_key]):
+        # Temporarily gate enterprise catalog api usage behind waffle sample
+        if waffle.sample_is_active(USE_ENTERPRISE_CATALOG):
+            if not EnterpriseCatalogApiClient().contains_content_items(self.uuid, [course_key]):
```

Comment:
Suggestion for a small engineering backlog ticket: a pattern I like to use for feature-gating is to create a `config.py` file that contains functions that map directly to whether a feature should be active.  So in this example, it'd be something like `def use_enterprise_catalog(*args, **kwargs): ...`.  Then all of your models, views, etc. can rely on that `config` module as a single interface for determining feature state, without worrying about the underlying switch/flag/settings that actually implement it (because we tend to change how features are toggled over the course of the release process).  I'll throw something in our tech-debt backlog if you think that makes sense.

Reply:
+1. This seems to be in line with the way the open edX WaffleFlag / WaffleSwitch are used in platform

Reply:
I think it's a good practice in general. In my mind this is a simple enough scenario that the config is over-engineering in my mind, but it doesn't matter much

Reply:
I think this makes sense to throw something onto our tech-debt backlog. Perhaps we could get around to it the next time we need to implement feature-toggling.